Automattic/woocommerce-payments
# Update/8581 replace intl tel input 2

Fixes #8581

#### Changes proposed in this Pull Request

* Use the new `PhoneNumberInput` component from `@woocommerce/components` instead of `intl-tel-input`
* Remove `intl-tel-input` from the package.json dependencies, remove all references.
* Bundle `PhoneNumberInput` since `@woocommerce/components` (`window.wc.components`) is not available in checkout.  See details in this comment: https://github.com/Automattic/woocommerce-payments/issues/8581#issuecomment-2246334922
* Chore: Update `watchPathIgnorePatterns` to prevent errors when running `npm run test:watch`

#### Testing instructions

##### Payment Settings

* Onboard with a new Stripe account or use one that doesn't have a support phone number set
* Go to WP Admin -> Payments -> Settings
* Verify that the Support phone number field is empty and that no errors appear
* Perform a chance in any other part of the settings (i.e. Toggle multi-currency or a payment method)
* Click Save Changes
* The changes should be saved successfully
* Update the support number to an invalid number
* The "Please enter a valid phone number error appears"
![image](https://github.com/user-attachments/assets/d02a78dc-e51c-451b-8386-a4c6b478ce9e)

* The "Save changes" button is disabled
* Use +1 0000000000 as the support phone number, no errors should appear and the Save Changes button is enabled.
* Use a valid phone number, no error should appear and the Save Changes button is enabled.
* Once all the previous tests are done, you can click "Save Changes", please note that once a number is saved, it cannot be removed, so you won't be able to repeat the previous steps unless you onboard with a new Stripe account.
* Once the phone has been saved, clear the Support Phone Number field. The error "Support phone number cannot be empty once it has been set before, please specify." appears and the Save Changes button is disabled.
![image](https://github.com/user-attachments/assets/7bfece22-8369-4324-8409-3a134e80ac96)

##### Checkout Page

* Enable WooPay
* As a guest shopper add a product to the cart
* Go to the checkout page (test both blocks and shortcode)
* Select the Securely save my information for 1-click checkout checkbox
* Make sure no error shows up below the input field and the place order button is not blocked
* Fill in all required fields but the WooPay phone number
* Click the place order button
* Make sure the phone input error shows up and the place order button is not blocked
* Type an invalid phone number
* The error should persist in the page, if you click place order it shouldn't go through
* Type a valid phone number
* Make sure the error message is gone
* Click the place order button
* Make sure the success page shows up


##### Payment Settings (non-US)

* Onboard or switch to a UK account
* Go to WP Admin -> Payments -> Settings
* Enter a valid UK phone number as the support number, click Save Changes
* Reload the page
* The Phone number should be set correctly and no validation errors should appear

-------------------


**Post merge**
